src/service/action-impl.js

```diff
@@ -121,6 +126,39 @@ export class ActionService {
   }
 
   /**
+   * Installs action handler for the specified element.
+   * @param {!Element} target
+   * @param {function(!ActionInvocation)} handler
+   */
+  installActionHandler(target, handler) {
+    const debugid = target.tagName + '#' + target.id;
+    assert(this.isAmpElement_(target), 'AMP element is expected: %s', debugid);
+
+    const currentQueue = target[ACTION_QUEUE_];
+    if (currentQueue) {
+      assert(Array.isArray(currentQueue), 'Expected queue to be an array: %s',
+          debugid);
+    }
+
+    // Override queue with the handler.
+    target[ACTION_QUEUE_] = {'push': handler};
```

Comment:
I see your point. However it's called in the separate event context and ideally we'd preserve it, e.g. when the action would show a popup. `setTimeout` could cause issues here. Deferred actions suffer from this - but there's nothing we can do about that. The only alternative here is to push the handler execution into a micro-task. But micro-tasks are not always implemented correctly everywhere and polyfills have the same issue - loosing event context.